tests/NuGet.Services.BasicSearchTests/TestSupport/LuceneInitializer.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Lucene.Net.Store;
+using NuGet.Indexing;
+
+namespace NuGet.Services.BasicSearchTests.TestSupport
+{
+    public class LuceneInitializer
+    {
+        private readonly BasicSearchTestSettings _settings;
+        private readonly DataServicePackageRepository _repository;
+
+        public LuceneInitializer(BasicSearchTestSettings settings)
+        {
+            _settings = settings;
+            _repository = new DataServicePackageRepository(_settings.PackageSource);
+        }
+
+        public async Task InitializeAsync()
+        {
+            await DownloadPackagesAsync();
+            CreateLuceneIndex();
+        }
+
+        private async Task DownloadPackagesAsync()
+        {
+            System.IO.Directory.CreateDirectory(_settings.PackageDirectory);
+            await Task.WhenAll(_settings.PackageVersions.Select(DownloadPackageAsync));
+        }
+
+        private void CreateLuceneIndex()
+        {
+            var directoryInfo = new DirectoryInfo(_settings.LuceneDirectory);
+            directoryInfo.Create();
+
+            using (var indexWriter = DocumentCreator.CreateIndexWriter(new SimpleFSDirectory(directoryInfo), true))
+            {
+                foreach (var version in _settings.PackageVersions)
+                {
+                    var metadata = GetPackageMetadata(version);
+                    var document = DocumentCreator.CreateDocument(metadata);
+                    indexWriter.AddDocument(document);
+                }
+
+                indexWriter.Commit();
+            }
+        }
+
+        private IDictionary<string, string> GetPackageMetadata(PackageVersionIdentifier version)
+        {
+            var path = GetPackagePath(version);
+            using (var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read))
+            {
+                var errors = new List<string>();
+                var metadata = PackageMetadataExtraction.MakePackageMetadata(fileStream, errors);
+
+                if (errors.Any())
+                {
+                    throw new InvalidOperationException(
+                        $"NuGet package for '{version.Id}' (version '{version.Version}') could be read for metadata. " +
+                        $"Errors: {string.Join(", ", errors)}");
+                }
+
+                return metadata;
+            }
+        }
+
+        private string GetPackagePath(PackageVersionIdentifier version)
+        {
+            return Path.Combine(_settings.PackageDirectory, $"{version.Id}.{version.Version}.nupkg".ToLower());
+        }
+        
+        private async Task DownloadPackageAsync(PackageVersionIdentifier version)
```

Comment:
I am not sure what you mean about this. Could you elaborate or point me to some code that does this?